src/braket/pennylane_plugin/braket_device.py

```diff
@@ -355,6 +366,33 @@ def apply(
 
         return circuit
 
+    def _parametrized_evolution_to_pulse_sequence(
+        self, op: ParametrizedEvolution, dev_wires
+    ) -> PulseSequence:
+        pulse = op.H.pulses[0]  # Assume only one `HardwarePulse`
+        amplitude = partial(pulse.amplitude, op.parameters[0])
+
+        frames = [self._device.frames[f"q{w}_drive"] for w in dev_wires]
+        sampling_periods = [f.port.dt * 1e9 for f in frames]  # seconds to nanoseconds
+
+        # TODO: Figure out unit of amplitude for `ArbitraryWaveform` and do conversion
+        amplitudes = [
+            [amplitude(t) * 1e9 for t in np.arange(op.t[0], op.t[1], dt)] for dt in sampling_periods
+        ]  # GHz to Hz
+
+        waveforms = [ArbitraryWaveform(amp) for amp in amplitudes]
+
+        pulse_sequence = PulseSequence().barrier(frames)
```

Comment:
But the barrier should I guess still be applied to all frames, since we don't know in PennyLane which qubits are being driven through cross-resonance. So I guess we need something like `drive_frames` and the `device_frames`. 

Do we know if there is any penalty/cost to applying the barrier to _everything_ every time?

Reply:
Yes excellent points!

Naively, I would think that putting the barrier on all device frames and just the ones we use should be equivalent. But there could be some dephasing going on. For that, however, I also dont know if putting a barrier and doing nothing helps / makes it worse / does nothing (thinking of that example where they manually shift the phases so they are back in sync in the rigetti example)

Reply:
I think the barrier is just telling all the relevant AWGs to reset their clocks by checking some reference input and synchronizing, but I'm not that confident. But if that's the case, I can't see the harm in all the AWGs doing it. Though if there is no reason to select specific ones, it's a bit odd the function isn't just `barrier()` with no args, and it synchs everything.

Reply:
(from this example notebook: https://github.com/aws/amazon-braket-examples/blob/main/examples/pulse_control/2_Bell_pair_with_pulses_OQC.ipynb)

Reply:
> Also, do we understand what is meant by `q0_q1_CR` in `.barrier([q0, q1, q0_q1_CR])`? That one wouldn't be included if we are just listing all the single qubit frames. Does that matter?

Indeed a bit odd if the CR frame is just an alias of driving one one qubit at the resonance freq. of the other 🤔 I'll add that to the list of open questions